### PR TITLE
Add a binary for generating the stats of the index

### PR DESCRIPTION
**Pull request by haiyanmeng**
1) how many kinds of objects are being customized?
2) how many times is every kind of object customized?
3) how many kustomization features are being customized?
4) how many times is every kustomization faeture customized?

### CODE REVIEW

File: api/internal/crawl/cmd/kustomize_stats/main.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	topKustomizeFeaturesPtr := flag.Int("kustomize-features", -1,
+		"the number of kustomize features to be listed according to their popularities." +
+			"(By default, all the features will be listed. " +
+			"If you only want to list the 10 most popular features, set the flag to 10.)")
+	flag.Parse()
+
+	ctx := context.Background()
+	idx, err := index.NewKustomizeIndex(ctx)
+	if err != nil {
+		fmt.Printf("Could not create an index: %v\n", err)
```
Comment (monopole) [inline api/internal/crawl/cmd/kustomize_stats/main.go]: log.Print("Could not..)
log.Fatal(err)


Presumably we'd want this to go to stderr, and want the process to exit with non-zero status

File: api/internal/crawl/cmd/kustomize_stats/main.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		}
+	}
+	return false
+}
+
+func main() {
+	topKindsPtr := flag.Int("kinds", -1,
+		"the number of kubernetes object kinds to be listed according to their popularities." +
+		"(By default, all the kinds will be listed. " +
+		"If you only want to list the 10 most popular kinds, set the flag to 10.)")
```
Comment (monopole) [inline api/internal/crawl/cmd/kustomize_stats/main.go]: Go let's one write paragraphs more directly:
```
topKindsPtr := flag.Int(
  "kinds", -1,
`the number of kubernetes object kinds to be listed according to their popularities.
By default, all the kinds will be listed.
If you only want to list the 10 most popular kinds, set the flag to 10.`)
```

File: api/internal/crawl/cmd/kustomize_stats/main.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+				kustomizationFilecount++
+				iterateArr(hit.Document.Identifiers, kustomizeIdentifiersMap)
+			}
+		}
+	}
+	if err := it.Err(); err != nil {
+		fmt.Printf("Error iterating: %v\n", err)
+	}
+
+	sortMapKeyByValueFunc := func(m map[string]int) []string {
```
Comment (monopole) [inline api/internal/crawl/cmd/kustomize_stats/main.go]: please just declare this outside main like `isKustomizationFile`
(main's a bit too long for my brain)

File: api/internal/crawl/cmd/kustomize_stats/main.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+			keys = append(keys, key)
+		}
+		// sort keys according to their values in the map m
+		sort.Slice(keys, func(i, j int) bool {return m[keys[i]] > m[keys[j]]})
+		return keys
+	}
+
+	fmt.Printf("The count of unique document IDs in the kustomize index: %d\n", len(ids))
+	fmt.Printf("There are %d documents in the kustomize index\n", count)
+	fmt.Printf("%d kinds of kubernetes objects are customized:\n", len(kindsMap))
```
Comment (monopole) [inline api/internal/crawl/cmd/kustomize_stats/main.go]:
```
	fmt.Printf(`
The count of unique document IDs in the kustomize index: %d
There are %d documents in the kustomize index
%d kinds of kubernetes objects are customized`,
		33, 44, 55)
```

File: api/internal/crawl/cmd/kustomize_stats/main.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+	ctx := context.Background()
+	idx, err := index.NewKustomizeIndex(ctx)
+	if err != nil {
+		fmt.Printf("Could not create an index: %v\n", err)
+		return
+	}
+
+	// count tracks the number of documents in the index
+	count := 0
```
Comment (monopole) [inline api/internal/crawl/cmd/kustomize_stats/main.go]: can main be written as 
```
main() {
  parse flags 
  stuff := computeStuff()
  report(stuff)
}
```

Comment (haiyanmeng) [reply api/internal/crawl/cmd/kustomize_stats/main.go]: Done

File: api/internal/crawl/cmd/kustomize_stats/main.go (hunk 4 above)
Comment (haiyanmeng) [reply api/internal/crawl/cmd/kustomize_stats/main.go]: Done

File: api/internal/crawl/cmd/kustomize_stats/main.go (hunk 3 above)
Comment (haiyanmeng) [reply api/internal/crawl/cmd/kustomize_stats/main.go]: Done

File: api/internal/crawl/cmd/kustomize_stats/main.go (hunk 2 above)
Comment (haiyanmeng) [reply api/internal/crawl/cmd/kustomize_stats/main.go]: done

File: api/internal/crawl/cmd/kustomize_stats/main.go (hunk 1 above)
Comment (haiyanmeng) [reply api/internal/crawl/cmd/kustomize_stats/main.go]: Done

**Pull request merged**